### PR TITLE
Fix database session entity class error

### PR DESCRIPTION
**Pull request by RuGa**
Fix allow use correct custom class and his mutators in table events instead base Entity class.

### CODE REVIEW

File: src/Http/Session/DatabaseSession.php [hunk 1]
```diff
@@ -155,7 +154,9 @@ public function write($id, $data): bool
         /** @var string $pkField */
         $pkField = $this->_table->getPrimaryKey();
         $record[$pkField] = $id;
-        $result = $this->_table->save(new Entity($record));
+        $entityClass = $this->_table->getEntityClass();
+        $entity = new $entityClass($record);
```
Comment (ADmad) [inline src/Http/Session/DatabaseSession.php:158]: We could just `$this->_table->newEntity($record)`

Comment (RuGa) [reply src/Http/Session/DatabaseSession.php:158]: Then need add "id" to $_accessible in Entity Class for mass assigned using newEntity() or patchEntity()

Comment (markstory) [reply src/Http/Session/DatabaseSession.php:158]: You can do that using the options to newEntity().

**Comment by ADmad** [comment]
@RuGa ping

**Comment by othercorey** [comment]
@RuGa Can you make the changes to use `newEntity()`?

**Comment by othercorey** [comment]
Implemented in https://github.com/cakephp/cakephp/pull/15345

**Pull request closed without merging**